lightning/src/ln/channelmanager.rs

```diff
@@ -7812,26 +7771,38 @@ where
 						return Err(MsgHandleErrInternal::from_chan_no_close(e, msg.common_fields.temporary_channel_id));
 					},
 					Ok(res) => res
-				};
+				})
+			},
+		};
 
-				let channel_type = channel.context.get_channel_type();
-				if channel_type.requires_zero_conf() {
-					return Err(MsgHandleErrInternal::send_err_msg_no_close("No zero confirmation channels accepted".to_owned(), msg.common_fields.temporary_channel_id));
-				}
-				if channel_type.requires_anchors_zero_fee_htlc_tx() {
-					return Err(MsgHandleErrInternal::send_err_msg_no_close("No channels with anchor outputs accepted".to_owned(), msg.common_fields.temporary_channel_id));
-				}
+		let channel_type = channel_phase.context().get_channel_type();
+		if channel_type.requires_zero_conf() {
+			return Err(MsgHandleErrInternal::send_err_msg_no_close("No zero confirmation channels accepted".to_owned(), common_fields.temporary_channel_id));
+		}
+		if channel_type.requires_anchors_zero_fee_htlc_tx() {
+			return Err(MsgHandleErrInternal::send_err_msg_no_close("No channels with anchor outputs accepted".to_owned(), common_fields.temporary_channel_id));
+		}
```

Comment:
Predates this PR, but I wonder why we don't do this earlier using `channel::channel_type_from_open_channel`. If we can, then it would make the rest of the code simpler since the `MessageSendEvent` can be push directly in the `match` against `msg` and `outbound_scid_alias` can be passed directly to the constructors.

Reply:
Yeah thanks, moving it above the `manually_accept_inbound_channels` check works just fine. We need it immediately in both cases either way. The zero conf and anchors checks I have left just after the manual acceptance check.